### PR TITLE
Fixed #1222 - 2.0: changed propagation

### LINKED ISSUE #1222: 2.0: changed propagation
* Remove changed notification from sub objects to parents
* No longer notifying parent objects when sub dictionaries or arrays are changed.
* This will allow to save the document without making changes; To detect whether the document is changed or not will be done by comparing encoded documents, which will be implemented in a separated commit.

### PR DESCRIPTION
**Pull request by hideki**
Ported from https://github.com/couchbase/couchbase-lite-ios/commit/845577034e94c3d8ead423d8cd799efb01190b23?diff=split

### CODE REVIEW

File: shared/src/main/java/com/couchbase/lite/Array.java [hunk 1]
```diff
@@ -439,40 +407,15 @@ private void set(int index, Object value, boolean isChange) {
     }
 
     private void setChanged() {
-        if (!changed) {
+        if (!changed)
             changed = true;
-            notifyChangeListeners();
-        }
         changedCount++;
```
Comment (pasin) [inline shared/src/main/java/com/couchbase/lite/Array.java]: 'changedCount ' No used?

File: shared/src/main/java/com/couchbase/lite/Array.java [hunk 2]
```diff
@@ -6,22 +6,19 @@
...
  */
-public class Array extends ReadOnlyArray implements ArrayInterface, ObjectChangeListener, FleeceEncodable {
+public class Array extends ReadOnlyArray implements ArrayInterface, FleeceEncodable {
     //---------------------------------------------
     // member variables
     //---------------------------------------------
     private List<Object> list = null;
-    Map<ObjectChangeListener, Integer> changeListeners = new HashMap<>();
     private boolean changed = false;
     private int changedCount = 0;
```
Comment (pasin) [inline shared/src/main/java/com/couchbase/lite/Array.java]: 'changedCount' no used?

File: shared/src/main/java/com/couchbase/lite/Dictionary.java [hunk 3]
```diff
@@ -561,31 +536,8 @@ private void set(String key, Object value, boolean isChange) {
     //  CHANGE
 
     private void setChanged() {
-        if (!changed) {
+        if (!changed)
             changed = true;
-            notifyChangeListeners();
-        }
         changedCount++;
```
Comment (pasin) [inline shared/src/main/java/com/couchbase/lite/Dictionary.java]: 'changedCount' no used?

File: shared/src/main/java/com/couchbase/lite/Array.java (hunk 2 above)
Comment (hideki) [inline shared/src/main/java/com/couchbase/lite/Array.java]: `changedCount` is still used by Iterator.
https://github.com/couchbase/couchbase-lite-android/blob/c029de8091bb931b1a53d2073ab954ca9b996904/shared/src/main/java/com/couchbase/lite/Array.java#L348-L368

File: shared/src/main/java/com/couchbase/lite/Array.java (hunk 1 above)
Comment (hideki) [inline shared/src/main/java/com/couchbase/lite/Array.java]: same with above my comment

File: shared/src/main/java/com/couchbase/lite/Dictionary.java (hunk 3 above)
Comment (hideki) [inline shared/src/main/java/com/couchbase/lite/Dictionary.java]: same with above my comment

**Pull request merged**